lib/internal/streams/writable.js

```diff
@@ -122,7 +122,13 @@ function WritableState(options, stream, isDuplex) {
   // Crypto is kind of old and crusty.  Historically, its default string
   // encoding is 'binary' so we have to make this configurable.
   // Everything else in the universe uses 'utf8', though.
-  this.defaultEncoding = (options && options.defaultEncoding) || 'utf8';
+  const defaultEncoding = options?.defaultEncoding;
+
+  if (defaultEncoding && !Buffer.isEncoding(defaultEncoding)) {
```

Comment:
defaultEncoding != null is better since if you pass false as defaultEncoding it bypasses isEncoding path and does not throw error.

Reply:
With null it goes into default `utf8`